### PR TITLE
Feature/39513

### PR DESCRIPTION
**Pull request by sa11616079**
* Create a new feature(feature/39513) branch according to the guidelines.
* Create a .editorconfig for Javascript and study about it.
* Create a CLIENT-SERVER.md file and write in brief about "how a request gets served".
* Create MERN-STACK.md file and write about MERN stack in detail.
* Create 12FACTOR-APP.md file and write about 12 Factor APP stack in detail.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+charset = utf-8
+
+trim_trailing_whitespace = true
+
+insert_final_newline = true
+
+end_of_line = lf
+
+max_line_length = off
```
Comment (SHUBHAM-JAIN-SS) [inline .editorconfig:16]: it is good practice to have a newline at the end of the file.

File: CLIENT-SERVER.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+**How a request gets served :**
+
+Computers connected to the web are called clients and servers.
+
+**1.** _Clients are the web user's internet-connected devices (for example, your computer connected to your Wi-Fi, or your phone connected to your mobile network) and web-accessing software available on those devices (usually a web browser like Firefox or Chrome)_.
+
+**2.** _Servers are computers that store webpages, sites, or apps. When a client device wants to access a webpage, a copy of the webpage is downloaded from the server onto the client machine to be displayed in the user's web browser_.
+
+_When you type HTTP request to your browser the first thing that happens is a Domain Name Server (DNS) matches your request to an IP address. Then the browser sends an HTTP request to the server and the server sends back an HTTP response. The browser begins rendering the HTML on the page while also requesting any additional resources such as CSS, JavaScript, images, etc. Each subsequent request completes a request/response cycle and is rendered in turn by the browser. Then once the page is loaded some sites e)(though not min will make further asynchronous requests_.
```
Comment (SHUBHAM-JAIN-SS) [inline CLIENT-SERVER.md:9]: please make the last line a bit clearer.

Comment (SHUBHAM-JAIN-SS) [inline CLIENT-SERVER.md:9]: please make the last line a bit clearer to understand.

**Comment by SHUBHAM-JAIN-SS** [comment]
Also, try to have less no of commits, standard practice is to have one commit per feature.

File: .editorconfig (hunk 1 above)
Comment (sa11616079) [reply .editorconfig:16]: ok sir

File: CLIENT-SERVER.md (hunk 2 above)
Comment (sa11616079) [reply CLIENT-SERVER.md:9]: Sir, The last line says that whenever we browse through some website or rather type the url of the website than we know that every website has some particular Ip address which is served by the domain name server(DNS) So whenever we type the url the DNS request the particular Ip address that we searching for, once the request is responded it comes back to the browser i.e the request we served in this process it extracts all the necessary things that are needed to present in the page we requesting for.

**Comment by sa11616079** [comment]
> Also, try to have less no of commits, standard practice is to have one commit per feature.

ok sir , I will take care of it.

**Pull request merged**